docs/build/guides/conventions/upgrading-contracts.mdx

```diff
@@ -0,0 +1,333 @@
+    let contract_id = env.register_contract_wasm(None, old_contract::Wasm);
+
+    let client = old_contract::Client::new(&env, &contract_id);
+    let admin = Address::random(&env);
+    client.init(&admin);
+
+    assert_eq!(1, client.version());
+
+    let new_wasm_hash = install_new_wasm(&env);
+
+    client.upgrade(&new_wasm_hash);
+    assert_eq!(2, client.version());
+
+    // new_v2_fn was added in the new contract, so the existing
+    // client is out of date. Generate a new one.
+    let client = new_contract::Client::new(&env, &contract_id);
+    assert_eq!(1010101, client.new_v2_fn());
+}
+
+```
+
+We first import Wasm files for both contracts:
+
+```rust
+mod old_contract {
+    soroban_sdk::contractimport!(
+        file =
+            "target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_old_contract.wasm"
+    );
+}
+
+mod new_contract {
+    soroban_sdk::contractimport!(
+        file = "../new_contract/target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_new_contract.wasm"
+    );
+}
+```
+
+We register the old contract, intialize it with an admin, and verify the version it returns. The note in the code below is important:
+
+```rust
+// Note that we use register_contract_wasm instead of register_contract
+// because the old contracts Wasm is expected to exist in storage.
+let contract_id = env.register_contract_wasm(None, old_contract::Wasm);
+
+let client = old_contract::Client::new(&env, &contract_id);
+let admin = Address::random(&env);
+client.init(&admin);
+
+assert_eq!(1, client.version());
+```
+
+We install the new contract's Wasm:
+
+```rust
+let new_wasm_hash = install_new_wasm(&env);
+```
+
+Then we run the upgrade, and verify that the upgrade worked:
+
+```rust
+client.upgrade(&new_wasm_hash);
+assert_eq!(2, client.version());
+```
+
+## Build the contract
+
+To build the contract `.wasm` files, run `stellar contract build` in both `upgradeable_contract/old_contract` and `upgradeable_contract/new_contract` in that order.
+
+Both `.wasm` files should be found in both contract `target` directories after building both contracts:
+
+```
+target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_old_contract.wasm
+```
+
+```
+target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_new_contract.wasm
+```
+
+## Run the contract
+
+If you have [`stellar-cli`] installed, you can invoke contract functions. Deploy the old contract and install the Wasm for the new contract.
+
+Navigate to to `upgradeable_contract/old_contract`
+
+```sh
+stellar contract deploy \
+  --wasm target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_old_contract.wasm  --source alice  \
+  --network testnet
+```
+
+- When you deploy a smart contract to a network, you need to specify an identity that will be used to sign the transactions. Change the `alice` to your own [identity].
+
+[identity]: ../../smart-contracts/getting-started/setup#configure-an-identity
```

Comment:
```suggestion
[identity]: ../../smart-contracts/getting-started/setup.mdx#configure-an-identity
```

I'm not super-sure of all the details/benefits, but docusaurus [recommends using the file extension in links](https://docusaurus.io/docs/2.x/markdown-features/links) (at the end of the page), so we try to do that whenever possible.